lib/bibdata_rs/src/theses/dataspace/community.rs

```diff
@@ -0,0 +1,91 @@
+// This module is responsible for interacting with Dataspace communities.
+// using the Dataspace JSON API.  Our Dataspace has a single community dedicated
+// to senior theses.
+
+use log::info;
+use serde::Deserialize;
+
+#[derive(Debug, Default, Deserialize)]
+struct Community {
+    pub id: Option<u32>,
+    pub handle: String,
+}
+
+#[derive(Debug, Default, Deserialize)]
+struct Collection {
+    pub id: Option<u32>,
+}
+
+/// The DSpace id of the community we're fetching content for.
+/// E.g., for handle '88435/dsp019c67wm88m', the DSpace id is 267
+pub fn get_community_id(
+    server: &str,
+    community_handle: &str,
+) -> Result<Option<u32>, reqwest::Error> {
+    let communities: Vec<Community> =
+        reqwest::blocking::get(format!("{}/communities/", server))?.json()?;
+    let theses_community = communities
+        .iter()
+        .find(|community| community.handle == community_handle);
+    // TODO!  Handle the case in which the json does not contain the community handle
+    Ok(theses_community.unwrap().id)
+}
+
+pub fn get_collection_list<T>(
+    server: &str,
+    community_handle: &str,
+    id_selector: T,
```

Comment:
Max was confused by this - would you add more comments to help future us understand?